packages/react/mixins/mixin.browser.js

```diff
@@ -45,10 +50,46 @@ class ReactMixin extends Mixin {
 }
 
 ReactMixin.strategies = {
-  bootstrap: parallel,
-  enhanceElement: compose,
-  fetchData: pipe,
-  render: override,
+  bootstrap: validate(parallel, ({ length }) => {
+    invariant(length === 0, 'bootstrap(): Received unexpected argument(s)');
+  }),
+  enhanceElement: validate(
+    compose,
+    ([element]) => {
+      invariant(
+        isValidElement(element),
+        'enhanceElement(): Received invalid React element'
+      );
+    },
+    (result) => {
+      invariant(
+        isValidElement(result),
+        'enhanceElement(): Returned invalid React element'
+      );
+    }
+  ),
+  fetchData: validate(
+    pipe,
+    ([data, element]) => {
+      invariant(
+        isPlainObject(data),
+        'fetchData(): Received invalid data object'
+      );
+      invariant(
+        isValidElement(element),
+        'fetchData(): Received invalid React element'
+      );
+    },
+    (result) => {
+      invariant(
+        isPlainObject(result),
+        'fetchData(): Returned invalid data object'
+      );
+    }
+  ),
+  render: validate(override, ({ length }) => {
+    invariant(length === 0, 'render(): Received unexpected argument(s)');
```

Comment:
This validation fires in our app (fresh `hops` setup without any custom mixings). When hot module patches are to be applied. `checkArgs` is called with `["./src/app.js"], ["./src/app.js"]` in that scenario.

Reply:
@Xiphe thanks, please see: https://github.com/untool/untool/pull/288

Reply:
just found it. thanks!